File: wagerproof-ios-native/WagerproofKit/Sources/WagerproofStores/NFLGameSheetStore.swift
import Foundation
import Observation
import WagerproofModels

/// Drives the NFL game detail bottom sheet. Mirrors the RN
/// `NFLGameSheetContext` (provider + `useNFLGameSheet` hook) but uses
/// SwiftUI's `.sheet(item:)` pattern — when `selectedGame != nil`, the sheet
/// is presented; nil-ing it dismisses.
///
/// `openGameSheet(_:)` matches RN's API. `closeGameSheet()` is provided for
/// imperative dismissal (e.g. when the sheet's close button calls it).
@Observable
@MainActor
public final class NFLGameSheetStore {
    public var selectedGame: NFLPrediction?

    public init() {}

    public func openGameSheet(_ game: NFLPrediction) {
        selectedGame = game
    }

    public func closeGameSheet() {
        selectedGame = nil
    }
}

File: wagerproof-android-native/core/stores/src/main/kotlin/com/wagerproof/core/stores/NFLGameSheetStore.kt
package com.wagerproof.core.stores

import androidx.compose.runtime.Stable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import com.wagerproof.core.models.NFLPrediction

/**
 * Drives the NFL game detail bottom sheet. Mirrors the iOS `NFLGameSheetStore`
 * — `selectedGame != nil` presents a [androidx.compose.material3.ModalBottomSheet];
 * nil-ing it dismisses.
 */
@Stable
class NFLGameSheetStore {
    var selectedGame by mutableStateOf<NFLPrediction?>(null)

    fun openGameSheet(game: NFLPrediction) {
        selectedGame = game
    }

    fun closeGameSheet() {
        selectedGame = null
    }
}
